src/box/memtx_space.c

```diff
@@ -991,8 +991,10 @@ memtx_space_check_format(struct space *space, struct tuple_format *format)
 	if (it == NULL)
 		return -1;
 
-	if (txn_check_singlestatement(txn, "space format check") != 0)
+	if (txn_check_singlestatement(txn, "space format check") != 0) {
+		iterator_delete(it);
 		return -1;
+	}
```

Comment:
Better move the check before the iterator construction.

Reply:
Fixed.